src/sw.prod.js

```diff
@@ -16,28 +16,51 @@
  *  limitations under the License
  *
  */
+/* global workbox */
+
 workbox.skipWaiting();
 workbox.clientsClaim();
 
-/* global workbox */
+/**
+ * Precache (cache ahead of time)
+ */
 
-// Precache static assets.
+// Precache assets generated by workbox-webpack-plugin
 workbox.precaching.precacheAndRoute(self.__precacheManifest || []);
 
-// Cache at runtime: partial navigation routes (fragments).
-const offlineNavigationResponse = `It seems you don't have
-  internet access at the moment.
-  Please try again later.`;
+/**
+ * Handle navigation requests
+ */
+
+workbox.routing.registerNavigationRoute('/app-shell');
+
+/**
+ * Runtime cache (cache when resource is accessed first time)
+ */
+
+// Cache partial navigation routes (fragments)
+/* eslint-disable max-len */
+const offlineNavigationResponse = `<div>
```

Comment:
Can this be included from a file? Even if it's injected into the service worker as a result of a build step or similar. I'm uncomfortable with the duplication here and it feels like this will be missed when the templates are updated.

Reply:
Not without substantial change to the build process. The offline message is not used in templates anywhere, so there is no duplication though. I also find it more readable to have it here, for instructional purposes (as this is a demo where we explain things - if we overcomplicate things with build, all learnings will be hidden away).

If you think it's worth our time though, I can extract it out in a separate next PR.

Reply:
Yeah, since it's effectively a partial it would be nice to have it with all the other partials. For follow PR is fine though.